python/src/uagents/protocol.py

```diff
@@ -37,6 +47,9 @@ def __init__(self, name: Optional[str] = None, version: Optional[str] = None):
         self._version = version or "0.1.0"
         self._canonical_name = f"{self._name}:{self._version}"
         self._digest = ""
+        self._use_storage = use_storage or store_message_history
```

Comment:
I added this extra flag, because some protocols might need to use storage for a completely different reason, and would not need to store messages, so this would provide a bit more flexibility.